.github/workflows/main.yml

```diff
@@ -0,0 +1,154 @@
+          # See https://github.com/conda-incubator/setup-miniconda#IMPORTANT
+          shell: bash -l {0}
+
+      steps:
+        - uses: actions/checkout@v2
+          with:
+            lfs: false
+            submodules: true
+
+        - name: Set up conda
+          uses: conda-incubator/setup-miniconda@v2
+          with:
+            auto-update-conda: true
+            python-version: ${{ matrix.python-version }}
+
+        # -----------------
+        # OS-specific setup
+        # -----------------
+        - name: Setup (Linux)
+          if: runner.os == 'Linux'
+          run: |
+            echo "CMAKE_GENERATOR=Unix Makefiles" >> $GITHUB_ENV
+            echo "LD_LIBRARY_PATH=$CONDA_PREFIX/lib" >> $GITHUB_ENV
+
+        - name: Setup (macOS)
+          if: runner.os == 'macOS'
+          # Newer versions of MacOS effectively block DYLD_LIBRARY_PATH being set (System Integrity Protection)
+          # Explicitly setting RPATH using `install_name_tool -add_rpath $CONDA_PREFIX/lib ./build/out/osqp_tester`
+          #   doesn't work either.
+          # Here we get around it by using a standard non-root location for .dylib files as a soft link
+          run: |
+            echo "CMAKE_GENERATOR=Unix Makefiles" >> $GITHUB_ENV
+            echo "DYLD_LIBRARY_PATH=$CONDA_PREFIX/lib" >> $GITHUB_ENV
+            ln -s $CONDA_PREFIX/lib ~/lib
+
+        - name: Setup (Windows)
+          if: runner.os == 'Windows'
+          run: |
+            echo "CMAKE_GENERATOR=MinGW Makefiles" >> $GITHUB_ENV
+            echo "$CONDA_PREFIX/Library/bin" >> $GITHUB_PATH
+        # -----------------
+
+        # Fetching mkl from the anaconda channel instead of defaults gives us the MKL runtime dynamic libraries
+        # as well (mkl_rt.<dll/so>), required during the runtime testing steps.
+        # MKL on Anaconda 2021.* seems to have inexplicably renamed mkl_rt.dll to mkl_rt.1.dll, so we insist on
+        # a version earlier than 2021
+        - name: Install python dependencies
+          run: |
+            conda install -c anaconda "mkl<2021" numpy scipy
+            conda info
+            conda list
+
+        - name: Build
+          run: |
+            cmake -S . -B build ${{ matrix.cmake_flags }} ${{ matrix.cmake_flags_extra }}
+            cmake --build build
+
+        - name: Test
+          run: |
+            ./build/out/osqp_tester
+          if: ${{ matrix.cmake_flags == '-DUNITTESTS=ON -DCOVERAGE=ON' }}
+
+        - name: Valgrid check
+          run: |
+            sudo apt-get install valgrind
+            valgrind --suppressions=.valgrind-suppress.supp --leak-check=full --gen-suppressions=all \
+              --track-origins=yes --error-exitcode=1 build/out/osqp_tester
+          if: ${{ runner.os == 'Linux' && matrix.cmake_flags == '-DUNITTESTS=ON -DCOVERAGE=ON' && matrix.cmake_flags_extra == '-DENABLE_MKL_PARDISO=OFF' }}
+
+        - name: Generate coverage
```

Comment:
Can we use the action that we do in the QDLDL repo here (https://github.com/osqp/qdldl/blob/715d8d560e80c131b62c7b695a4ac8e52a9e2d3c/.github/workflows/ci.yml#L50)? I think it would be good to make the workflows as similar as possible so that the maintenance burden is less.

Reply:
Sure - using that action instead now.